animation_nodes/nodes/matrix/c_utils.pyx

```diff
@@ -137,6 +137,45 @@ cdef void scaleFromMatrix(Vector3 *scale, Matrix4 *m):
         scale.y *= -1.0
         scale.z *= -1.0
 
+def extractMatricesXBasis(Matrix4x4List matrices):
+    cdef Vector3DList normals = Vector3DList(length = len(matrices))
+    cdef Matrix4 *_matrices = matrices.data
+    cdef Vector3 *_normals = normals.data
+    cdef Py_ssize_t i
+
+    for i in range(len(normals)):
+        _normals[i].x = _matrices[i].a11
+        _normals[i].y = _matrices[i].a21
+        _normals[i].z = _matrices[i].a31
+
+    return normals
+
+def extractMatricesYBasis(Matrix4x4List matrices):
+    cdef Vector3DList normals = Vector3DList(length = len(matrices))
+    cdef Matrix4 *_matrices = matrices.data
+    cdef Vector3 *_normals = normals.data
+    cdef Py_ssize_t i
+
+    for i in range(len(normals)):
+        _normals[i].x = _matrices[i].a12
+        _normals[i].y = _matrices[i].a22
+        _normals[i].z = _matrices[i].a32
+
+    return normals
+
+def extractMatricesZBasis(Matrix4x4List matrices):
+    cdef Vector3DList normals = Vector3DList(length = len(matrices))
```

Comment:
Better call this `bases`, normals doesn't make sense here.

Reply:
Okay.

Reply:
Done.